payjoin/src/send/v2/mod.rs

```diff
@@ -32,12 +32,29 @@ use super::error::BuildSenderError;
 use super::*;
 use crate::hpke::{decrypt_message_b, encrypt_message_a, HpkeSecretKey};
 use crate::ohttp::{ohttp_decapsulate, ohttp_encapsulate};
+use crate::persist::{Persister, Value};
 use crate::send::v1;
 use crate::uri::{ShortId, UrlExt};
 use crate::{HpkeKeyPair, HpkePublicKey, IntoUrl, OhttpKeys, PjUri, Request};
 
 mod error;
 
+#[derive(Debug)]
+pub struct NewSender {
+    pub(crate) v1: v1::Sender,
+    pub(crate) reply_key: HpkeSecretKey,
+}
+
+impl NewSender {
+    pub fn persist<P: Persister<Sender>>(
+        &self,
+        persister: &mut P,
+    ) -> Result<P::Token, ImplementationError> {
+        let sender = Sender { v1: self.v1.clone(), reply_key: self.reply_key.clone() };
+        Ok(persister.save(sender)?)
+    }
+}
+
```

Comment:
It'd be more readable for this type to be defined after SenderBuilder in its place in order of the state machinery

Reply:
Fixed